plugins/base/src/main/kotlin/transformers/pages/annotations/SinceKotlinTransformer.kt

```diff
@@ -1,93 +1,171 @@
-                functions = functions.map { it.transform() as DFunction },
-                properties = properties.map { it.transform() as DProperty }
+                documentation = appendSinceKotlin(versions),
+                classlikes = classlikes.map { it.transform(versions) as DClasslike },
+                functions = functions.map { it.transform(versions) as DFunction },
+                properties = properties.map { it.transform(versions) as DProperty }
             )
+
             is DObject -> copy(
-                documentation = appendSinceKotlin(),
-                classlikes = classlikes.map { it.transform() as DClasslike },
-                functions = functions.map { it.transform() as DFunction },
-                properties = properties.map { it.transform() as DProperty }
+                documentation = appendSinceKotlin(versions),
+                classlikes = classlikes.map { it.transform(versions) as DClasslike },
+                functions = functions.map { it.transform(versions) as DFunction },
+                properties = properties.map { it.transform(versions) as DProperty }
             )
+
+            is DTypeAlias -> copy(
+                documentation = appendSinceKotlin(versions)
+            )
+
             is DAnnotation -> copy(
-                documentation = appendSinceKotlin(),
-                classlikes = classlikes.map { it.transform() as DClasslike },
-                functions = functions.map { it.transform() as DFunction },
-                properties = properties.map { it.transform() as DProperty }
+                documentation = appendSinceKotlin(versions),
+                classlikes = classlikes.map { it.transform(versions) as DClasslike },
+                functions = functions.map { it.transform(versions) as DFunction },
+                properties = properties.map { it.transform(versions) as DProperty }
             )
+
             is DFunction -> copy(
-                documentation = appendSinceKotlin()
+                documentation = appendSinceKotlin(versions)
             )
+
             is DProperty -> copy(
-                documentation = appendSinceKotlin()
+                documentation = appendSinceKotlin(versions)
             )
+
             is DParameter -> copy(
-                documentation = appendSinceKotlin()
+                documentation = appendSinceKotlin(versions)
             )
+
             else -> this.also { context.logger.warn("Unrecognized documentable $this while SinceKotlin transformation") }
         }
+    }
+
+    private fun List<Annotations.Annotation>.findSinceKotlinAnnotation(): Annotations.Annotation? =
+        this.find { it.dri.packageName == "kotlin" && it.dri.classNames == "SinceKotlin" }
+
+    private fun Documentable.getVersion(sourceSet: DokkaConfiguration.DokkaSourceSet): Version {
+        val annotatedVersion =
+            safeAs<WithExtraProperties<Documentable>>()?.extra?.get(Annotations)?.directAnnotations?.get(sourceSet)
```

Comment:
As mentioned previously, I believe it can be substituted with an existing function `this.annotations()[sourceSet]`